app/helpers/lookup_keys_helper.rb

```diff
@@ -72,35 +72,41 @@ def overridable_lookup_keys(klass, host)
     klass.class_params.override.where(:environment_classes => {:environment_id => host.environment}) + klass.lookup_keys
   end
 
-  def hostgroup_key_with_diagnostic(hostgroup, key)
-    value, origin = hostgroup.inherited_lookup_value key
-    original_value = key.value_before_type_cast value
-    diagnostic_helper = popover('', _("<b>Description:</b> %{desc}<br><b>Type:</b> %{type}<br> <b>Matcher:</b> %{matcher}") % { :desc => key.description, :type => key.key_type, :matcher => origin}, :data => { :placement => 'top' })
-    parameter_value_content("value_#{key.key}", original_value, :popover => diagnostic_helper)
-  end
+  def key_with_diagnostic(obj, key, index, lookup_value, value_hash = nil)
```

Comment:
I find this considerably harder to read than what we had before, one method for hostgroup and another for hosts. What do we gain out of merging them if we have to constantly check what object are we using inside?